### ISSUE TITLE
Section page break

### ISSUE THREAD

**Issue by Vartkat** [OCA/sale-reporting#71]
Hello,

Perhaps I'm missing something, but in Odoo V11, layout sections could get a page break before. 
Sometime, a section begins at the end of a page and gets another part on the next page, it vene happens that a section is on one page and only its subtotal line on the next one, or the other way around, a section title is on a page and the section content on the next one.
Wouldn't it be nice to get back the page-break feature ?

Thanks

Regards

**Comment by pedrobaeza** [comment]
Well, yes, you can add a module for re-adding that feature if you see it useful.

**Comment by Vartkat** [comment]
Unfortunately my skills doesn't go that far. I tried to do it with a Qweb inheritance but as line sections are not materialized (I mean are not tagged as a div for example) in HTML, I can't introduce a page-break:avoid.
My knowledge go that far, I wish I could and I tried. Avoiding page breaks in a section are very usefull to get a nice layout, i.e. if a section begins on page one and ends on page 2 it's nicer to have it begining on page 2.

**Comment by pedrobaeza** [comment]
OK, now we keep this if someone wants to do it.

**Comment by Vartkat** [comment]
Just a dummy question...

I was reading all your code and I was wondering... As you're adressing sections in your code, shouldn't it be as easy as inserting a "page-break-inside:avoid;" somewhere ? 
I don't mean to modify your code but if you tell me where to introduce that, perhaps I can inherit your templates and do it just for me.

**Comment by pedrobaeza** [comment]
I don't know which code are you referring to, but you need to make that configurable via UI, so it's not that simple.

**Comment by Vartkat** [comment]
OK I'll wait till someone with the skills take a look at it. Thanks a lot for your quick answers.

**Comment by Vartkat** [comment]
I'm renewing my request if ever someone could look into it. 

I tried to understand the code, think I quite did but that doesn't allow me to write some additioal code.
As I know how to inherit a report I tried to but as there's no HTML tag for enclosing a section (not even a <div>), I can't add a "page-break avoid".

What I would like is a third icon (next to the two added by this module) allowing me to introduce a page break after a section so that, when a section runs on two pages, I can manage to put it on one page (the next page by adding a page break to the section just before).
That could also be usefull when details of an invoice are on one page and total on the next one (ugly). With page-break, I should be able to spread my sections on several pages so that the layout looks fine.

Thanks for reading and helping
Regards

**Comment by Vartkat** [comment]
@pedrobaeza  @ernestotejeda 

I followed your advice and after reading many times your code I jumped into coding even if I don't know how to write python. 
I succeeded in adding a new value 'add_page_break' in the sale_order.py file. 
I succeeded in adding an icon next to the eye and plus icons (in the sale_views.xml). 
I verified that the mechanic behind the icon works. 
It works, I mean when I toggle the icon the value in the database is updated.

Now I'm struggling since a few days with th XPath (in sale_order_report_templates.xml) to insert my page_break before the section title line or after the section sub-total line.

The XPath I found is not satisfactory. I don't know why but inserting the page-break does a great whyte space in the page but the next section is not on the next page, instead it's title line is down the first page and then it's next lines are on the second page.

If ever an experienced devel could take a look at my code and point me in the right direction, that would be very kind.
https://github.com/Vartkat/sale-reporting

Thanks for your help

Regards


![image](https://user-images.githubusercontent.com/1696899/69107051-7728d100-0a70-11ea-990a-d37c20e7b318.png)
